sklearn/gaussian_process/_gpr.py

```diff
@@ -485,6 +490,15 @@ def sample_y(self, X, n_samples=1, random_state=0):
             Values of n_samples samples drawn from Gaussian process and
             evaluated at query points.
         """
+        # TODO(1.5): Default updates to `None`
+        if random_state == "warn":
+            warnings.warn(
+                "The default value of random_state will change from"
+                + " 0 to None in 1.5",
```

Comment:
We should tell users how to silence the warning. Something like the following text:

```suggestion
                "The default value of random_state will change from"
                " 0 to None in version 1.5. Explicitly set a value for `random_state` "
                "to silence this warning",
```